packages/mattermost-redux/src/selectors/entities/channels.ts

```diff
@@ -506,20 +513,24 @@ export const getUnreads: (state: GlobalState) => {
                 otherUserId = getUserIdFromChannelName(currentUserId, channel.name);
 
                 if (users[otherUserId] && users[otherUserId].delete_at === 0) {
-                    mentionCountForCurrentTeam += m.mention_count;
+                    mentionCountForCurrentTeam += (collapsed ? m.mention_count_root : m.mention_count);
+                }
+            } else if (channel.delete_at === 0) {
+                if (m.mention_count > 0) {
```

Comment:
I thought our ESLint rules would complain about this pattern, but I guess it's fine with it as long as the outer one is an else-if. Regardless of ESLint, I definitely prefer this since it makes it easier to understand the thinking here.